packages/core/content-releases/server/src/services/release-action.ts

```diff
@@ -243,16 +243,32 @@ const createReleaseActionService = ({ strapi }: { strapi: Core.Strapi }) => {
         return acc;
       }, []);
 
-      const contentTypeModelsMap = contentTypeUids.reduce(
-        (
-          acc: { [key: ReleaseAction['contentType']]: Struct.ContentTypeSchema },
+      const workflowsService = strapi.plugin('review-workflows').service('workflows');
+
+      const contentTypeModelsMap = await contentTypeUids.reduce(
+        async (
+          accPromise: Promise<{
+            [key: ReleaseAction['contentType']]: Struct.ContentTypeSchema & {
+              requiredStageToPublish?: number;
+            };
```

Comment:
could we use the type in the contract here? the one that merges the content type schema and the requriedStageToPublish, that will be less types to maintain